vyos/vyos-1x
# T5586: Disable by default SNMP for Keeplived VRRP service

## Change Summary

`AgentX` does not work stable.
From time to time we see the system service crashing/degrading if something is wrong with SNMP from util `net-snmp`.
We should disable it by default and enable it only if configured.

## Types of changes

- [x] Other (please describe): Change by default don't use SNMP for Keepalived

## Related Task(s)

* https://vyos.dev/T5586

## Component(s) name

vrrp
## Proposed changes


## How to test
Configure VRRP, be sure the `--snmp` flag is not set
```
set high-availability vrrp group GRP01 address 192.0.2.1/32
set high-availability vrrp group GRP01 interface 'eth1'
set high-availability vrrp group GRP01 no-preempt
set high-availability vrrp group GRP01 vrid '10'
commit


vyos@r4# ps ax | grep keepalived | grep -ve "grep"
   3906 ?        Ss     0:00 /usr/sbin/keepalived --use-file /run/keepalived/keepalived.conf --pid /run/keepalived/keepalived.pid --dont-fork
   3919 ?        S      0:00 /usr/sbin/keepalived --use-file /run/keepalived/keepalived.conf --pid /run/keepalived/keepalived.pid --dont-fork
   3920 ?        Sl     0:00 python3 /usr/libexec/vyos/system/keepalived-fifo.py /run/keepalived/keepalived_notify_fifo
[edit]
vyos@r4# 

```
Enable snmp and check `--snmp` flag
```
vyos@r4# set high-availability vrrp snmp 
[edit]
vyos@r4# commit
[edit]
vyos@r4# ps ax | grep keepalived | grep -ve "grep"
   5378 ?        Ss     0:00 /usr/sbin/keepalived --use-file /run/keepalived/keepalived.conf --pid /run/keepalived/keepalived.pid --dont-fork --snmp
   5380 ?        S      0:00 /usr/sbin/keepalived --use-file /run/keepalived/keepalived.conf --pid /run/keepalived/keepalived.pid --dont-fork --snmp
   5382 ?        Sl     0:00 python3 /usr/libexec/vyos/system/keepalived-fifo.py /run/keepalived/keepalived_notify_fifo
[edit]
vyos@r4# 
```
## Smoketest
```
vyos@r4:~$ /usr/libexec/vyos/tests/smoke/cli/test_ha_vrrp.py
test_01_default_values (__main__.TestVRRP.test_01_default_values) ... ok
test_02_simple_options (__main__.TestVRRP.test_02_simple_options) ... ok
test_03_sync_group (__main__.TestVRRP.test_03_sync_group) ... ok
test_04_exclude_vrrp_interface (__main__.TestVRRP.test_04_exclude_vrrp_interface) ... ok

----------------------------------------------------------------------
Ran 4 tests in 24.542s

OK
```
## Checklist:
